packages/devtools_app/lib/src/shared/table.dart

```diff
@@ -464,29 +465,23 @@ class TreeTableState<T extends TreeNode<T>> extends State<TreeTable<T>>
     addAutoDisposeListener(selectionNotifier, () {
       setState(() {
         final node = selectionNotifier.value.node;
-        //offset will need to be modified with new UI table changes
-        final offset = defaultRowHeight;
-        expandParents(node?.parent);
+        if (node == null) return;
+        expandParents(node);
 
-        node?.select();
-
-        final searchCondition = (T n) => n.isSelected;
-        final int selectedDisplayRow = node!.childCountToMatchingNode(
+        final searchCondition = (T n) {
+          return n == node;
+        };
+        final int selectedRowIndex = dataRoots[0].childCountToMatchingNode(
```

Comment:
We should start searching for the node from the root node, not the selected node. Note: This needs to be changed to loop through all the dataRoot nodes (not just the first one)